hedera-node/src/main/java/com/hedera/services/utils/accessors/PlatformTxnAccessor.java

```diff
@@ -22,72 +22,273 @@
  * whose contents is <i>supposed</i> to be a Hedera Services gRPC {@link Transaction}. (The constructor of this
  * class immediately tries to parse the {@code byte[]} contents of the txn, and propagates any protobuf
  * exceptions encountered.)
  */
-public class PlatformTxnAccessor extends SignedTxnAccessor {
+public class PlatformTxnAccessor implements TxnAccessor {
 	private final SwirldTransaction platformTxn;
-	private final AliasManager aliasManager;
+	private TxnAccessor delegate;
 
 	private RationalizedSigMeta sigMeta = null;
 
-	public PlatformTxnAccessor(SwirldTransaction platformTxn, final AliasManager aliasManager)
+	protected PlatformTxnAccessor(final TxnAccessor delegate,
+			SwirldTransaction platformTxn)
 			throws InvalidProtocolBufferException {
-		super(platformTxn.getContents());
 		this.platformTxn = platformTxn;
-		this.aliasManager = aliasManager;
+		this.delegate = delegate;
 	}
 
-	/**
-	 * Convenience static factory for a txn whose {@code byte[]} contents are <i>certain</i>
-	 * to be a valid serialized gRPC txn.
-	 *
-	 * @param platformTxn
-	 * 		the txn to provide accessors for.
-	 * @param aliasManager
-	 * @return an initialized accessor.
-	 */
+	public static PlatformTxnAccessor from(final TxnAccessor delegate, final SwirldTransaction platformTxn)
+			throws InvalidProtocolBufferException {
+		try {
+			return new PlatformTxnAccessor(delegate, platformTxn);
+		} catch (InvalidProtocolBufferException ignore) {
+			throw new IllegalStateException("Accessor construction must get valid gRPC bytes!");
+		}
+	}
 
-	public static PlatformTxnAccessor uncheckedAccessorFor(SwirldTransaction platformTxn,
-			final AliasManager aliasManager) {
+	public static PlatformTxnAccessor from(final SwirldTransaction platformTxn)
+			throws InvalidProtocolBufferException {
 		try {
-			return new PlatformTxnAccessor(platformTxn, aliasManager);
+			return new PlatformTxnAccessor(SignedTxnAccessor.from(platformTxn.getContentsDirect()),
+					platformTxn);
 		} catch (InvalidProtocolBufferException ignore) {
-			throw new IllegalStateException("Unchecked accessor construction must get valid gRPC bytes!");
+			throw new IllegalStateException("Accessor construction must get valid gRPC bytes!");
 		}
 	}
 
-	@Override
 	public SwirldTransaction getPlatformTxn() {
 		return platformTxn;
 	}
 
-	@Override
 	public void setSigMeta(RationalizedSigMeta sigMeta) {
 		this.sigMeta = sigMeta;
 	}
 
-	@Override
 	public RationalizedSigMeta getSigMeta() {
 		return sigMeta;
 	}
 
-	protected EntityNum unaliased(AccountID grpcId) {
-		return aliasManager.unaliased(grpcId);
+	@Override
+	public int sigMapSize() {
+		return delegate.sigMapSize();
+	}
+
+	@Override
+	public int numSigPairs() {
+		return delegate.numSigPairs();
+	}
+
+	@Override
+	public SignatureMap getSigMap() {
+		return delegate.getSigMap();
+	}
+
+	@Override
+	public void setExpandedSigStatus(final ResponseCodeEnum status) {
```

Comment:
Does it make sense to have this method on the `TxnAccessor` interface? (Unless the accessor was constructed from a `SwirldsTransaction`, it won't have gone through the `expandSignatures()` -> `handleTransaction()` lifecycle.)